File: src/main/java/dev/zprestige/fire/module/player/packetmine/DamageBlockListener.java
package dev.zprestige.fire.module.player.packetmine;

import dev.zprestige.fire.newbus.EventListener;
import dev.zprestige.fire.newbus.events.BlockInteractEvent;
import dev.zprestige.fire.util.impl.BlockUtil;
import net.minecraft.init.Blocks;

public class DamageBlockListener extends EventListener<BlockInteractEvent.DamageBlock, PacketMine> {

    public DamageBlockListener(final PacketMine packetMine){
        super(BlockInteractEvent.DamageBlock.class, packetMine);
    }

    @Override
    public void invoke(final Object object){
        final BlockInteractEvent.DamageBlock event = (BlockInteractEvent.DamageBlock) object;
        if (module.activePos != null) {
            module.abort(module.activePos, module.facing);
        }
        if (!BlockUtil.getState(event.getPos()).equals(Blocks.BEDROCK)) {
            module.initiateBreaking(event.getPos(), event.getFacing());
            event.setCancelled();
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/player/packetmine/DamageBlockListener.kt
package dev.zprestige.fire.module.player.packetmine

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.BlockInteractEvent.DamageBlock
import dev.zprestige.fire.util.impl.BlockUtil
import net.minecraft.init.Blocks

class DamageBlockListener(packetMine: PacketMine) : EventListener<DamageBlock, PacketMine>(DamageBlock::class.java, packetMine) {
    
    override fun invoke(e: Any) {
        val event = e as DamageBlock
        if (module.nullCheck()) {
            if (module.activePos != null) {
                module.abort(module.activePos, module.facing)
            }
            if (BlockUtil.getState(event.pos) != Blocks.BEDROCK) {
                module.initiateBreaking(event.pos, event.facing)
                event.setCancelled()
            }
        }
    }
}